addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -534,7 +527,10 @@ import wUtils from '@website/js/utils';
                         if (!date || !date.isValid) {
                             return true;
                         }
-                    } else if (input.type === "file" && !self.isFileInputValid(input)) {
+                    } else if (input.type === "file") {
+                        return self.isFileInputValid(input);
+                    } else if (self._requirementFunction(field) === false) {
```

Comment:
@msh-odoo Yes, it returns undefined whenever there is no comparator on the field. This happens because the ```_requirementFunction``` relies on ```_compareTo```, and every return statement in ```_compareTo``` is within a switch case. When the comparator is null, none of the return statements are executed, causing the function to return undefined. As a result, fields are incorrectly marked as invalid, even though some of them don’t actually require validation.